install_script_op_worker.sh

```diff
@@ -0,0 +1,553 @@
+
+Troubleshooting and basic usage information for the $nice_flavor are available at:
+
+    https://docs.datadoghq.com/observability_pipelines/\n\033[0m\n"
+
+    ERROR_MESSAGE=$SAVED_ERROR_MESSAGE
+
+    if ! tty -s; then
+      fallback_msg
+      exit 1;
+    fi
+
+    if [ "$site" == "ddog-gov.com" ]; then
+      fallback_msg
+      exit 1;
+    fi
+
+    while true; do
+        read -t 60 -p  "Do you want to send a failure report to Datadog (including $logfile)? (y/[n]) " -r yn || on_read_error
+        case $yn in
+          [Yy]* )
+            get_email
+            if [[ -n "$isEmailValid" ]]; then
+              report
+            fi
+            fallback_msg
+            break;;
+          [Nn]*|"" )
+            fallback_msg
+            break;;
+          * )
+            printf "Please answer yes or no.\n"
+            ;;
+        esac
+    done
+}
+trap on_error ERR
+
+# OPW doesn't have a public changelog
+function verify_agent_version(){
+    local ver_separator="$1"
+    if [ -z "$agent_version_custom" ]; then
+        ERROR_MESSAGE="Specified version not found: $worker_major_version.$worker_minor_version"
+        echo -e "
+  \033[33mWarning: $ERROR_MESSAGE
+\033[0m"
+        fallback_msg
+        exit 1;
+    else
+        worker_flavor+="$ver_separator$agent_version_custom"
+    fi
+}
+
+echo -e "\033[34m\n* Datadog Observability Pipelines Worker install script v${install_script_version}\n\033[0m"
+
+site=
+if [ -n "$DD_SITE" ]; then
+    site="$DD_SITE"
+fi
+
+apikey=
+if [ -n "$DD_API_KEY" ]; then
+    apikey=$DD_API_KEY
+fi
+
+op_pipeline_id=
+if [ -n "$DD_OP_PIPELINE_ID" ]; then
+    op_pipeline_id=$DD_OP_PIPELINE_ID
+fi
+
+no_start=
+if [ -n "$DD_INSTALL_ONLY" ]; then
+    no_start=true
+fi
+
+if [ -n "$DD_REPO_URL" ]; then
+    repository_url=$DD_REPO_URL
+elif [ -n "$REPO_URL" ]; then
+    echo -e "\033[33mWarning: REPO_URL is deprecated and might be removed later (use DD_REPO_URL instead).\033[0m"
+    repository_url=$REPO_URL
+else
+    repository_url="datadoghq.com"
+fi
+
+if [ -n "$TESTING_KEYS_URL" ]; then
+  keys_url=$TESTING_KEYS_URL
+else
+  keys_url="keys.datadoghq.com"
+fi
+
+if [ -n "$TESTING_YUM_URL" ]; then
+  yum_url=$TESTING_YUM_URL
+else
+  yum_url="yum.${repository_url}"
+fi
+
+# We turn off `repo_gpgcheck` for custom REPO_URL, unless explicitly turned
+# on via DD_RPM_REPO_GPGCHECK.
+# There is more logic for redhat/suse in their specific code branches below
+rpm_repo_gpgcheck=
+if [ -n "$DD_RPM_REPO_GPGCHECK" ]; then
+    rpm_repo_gpgcheck=$DD_RPM_REPO_GPGCHECK
+else
+    if [ -n "$REPO_URL" ] || [ -n "$DD_REPO_URL" ]; then
+        rpm_repo_gpgcheck=0
+    fi
+fi
+
+if [ -n "$TESTING_APT_URL" ]; then
+  apt_url=$TESTING_APT_URL
+else
+  apt_url="apt.${repository_url}"
+fi
+
+report_failure_url="https://api.datadoghq.com/agent_stats/report_failure"
+if [ -n "$DD_SITE" ]; then
+    report_failure_url="https://api.${DD_SITE}/agent_stats/report_failure"
+fi
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏ I'll look for @bkabrda's feedback here. It seems like we'll want to separate out install failures for the OPW.

Reply:
Ah, there was some feedback around this captured outside of the PR review - I believe the endpoint will remain the same for now, but the payload/handler adjusted here and in dogweb to account for the agent vs opw difference.

Reply:
Ah, gotcha 👍

Reply:
Added some additional context in the summary.